samples/containers/poetry/src/main/java/com/squareup/sample/poetry/PoemWorkflow.kt

```diff
@@ -1,111 +1,274 @@
 package com.squareup.sample.poetry
 
+import com.squareup.sample.container.overviewdetail.OverviewDetailContainer
 import com.squareup.sample.container.overviewdetail.OverviewDetailScreen
 import com.squareup.sample.poetry.PoemWorkflow.Action.ClearSelection
 import com.squareup.sample.poetry.PoemWorkflow.Action.HandleStanzaListOutput
 import com.squareup.sample.poetry.PoemWorkflow.Action.SelectNext
 import com.squareup.sample.poetry.PoemWorkflow.Action.SelectPrevious
-import com.squareup.sample.poetry.PoemWorkflow.ClosePoem
+import com.squareup.sample.poetry.PoemWorkflow.Config
+import com.squareup.sample.poetry.PoemWorkflow.Output
+import com.squareup.sample.poetry.PoemWorkflow.Output.ClosePoem
+import com.squareup.sample.poetry.PoemWorkflow.Output.Loading
+import com.squareup.sample.poetry.PoemWorkflow.PoemRenderingWithLoading
+import com.squareup.sample.poetry.PoemWorkflow.State
+import com.squareup.sample.poetry.PoemWorkflow.State.ComplexCall
+import com.squareup.sample.poetry.PoemWorkflow.State.Initializing
+import com.squareup.sample.poetry.PoemWorkflow.State.Selected
 import com.squareup.sample.poetry.StanzaWorkflow.Output.CloseStanzas
 import com.squareup.sample.poetry.StanzaWorkflow.Output.ShowNextStanza
 import com.squareup.sample.poetry.StanzaWorkflow.Output.ShowPreviousStanza
 import com.squareup.sample.poetry.StanzaWorkflow.Props
 import com.squareup.sample.poetry.model.Poem
+import com.squareup.sample.poetry.views.BlankScreen
 import com.squareup.workflow1.Snapshot
 import com.squareup.workflow1.StatefulWorkflow
+import com.squareup.workflow1.Worker
 import com.squareup.workflow1.WorkflowAction
 import com.squareup.workflow1.WorkflowAction.Companion.noAction
+import com.squareup.workflow1.action
 import com.squareup.workflow1.parse
+import com.squareup.workflow1.runningWorker
+import com.squareup.workflow1.ui.AndroidViewRendering
+import com.squareup.workflow1.ui.BuilderViewFactory
+import com.squareup.workflow1.ui.ViewFactory
 import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
 import com.squareup.workflow1.ui.backstack.BackStackScreen
 import com.squareup.workflow1.ui.backstack.toBackStackScreen
+import kotlinx.coroutines.delay
 
 typealias SelectedStanza = Int
 
 /**
  * Renders a [Poem] as a [OverviewDetailScreen], whose overview is a [StanzaListRendering]
  * for the poem, and whose detail traverses through [StanzaRendering]s.
  */
-object PoemWorkflow : StatefulWorkflow<Poem, SelectedStanza, ClosePoem, OverviewDetailScreen>() {
-  object ClosePoem
-  public const val NO_SELECTED_STANZA = -1
+object PoemWorkflow : StatefulWorkflow<Config, State, Output, PoemRenderingWithLoading>() {
+  const val NO_SELECTED_STANZA = -1
+
+  data class Config(
```

Comment:
I'm not happy with all this complexity leaking into the poetry sample itself. This makes it much less useful as an example of how to deal with complex composition. If there's no better alternative, I'd rather see it forked.

Reply:
Perhaps you could take a manual DI approach? e.g. Make `PoemWorkflow` an interface with the original code as its default implementation, and give `PoemActivity` a component interface like `TicTacToeComponent`.

Need to be thoughtful about whether that too would muddy the example too much. At least it would be more similar to how a production app would be built.

Reply:
Another nice effect of the DI approach is that it paves the way to multiple scenarios in the future, w/o requiring more and more branching here.